From 22e512db2b22d7f9e7733950262465b2315f5666 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 13 Mar 2024 07:45:54 +0200
Subject: [PATCH] [FSW-402] fs1: compilation fixes

This patch fixes compilation issues in current state of code.
This also removes fields that will be modified later due
to change from erom to recovery img.

Change-Id: I1ac022ecd52799dc2446657b7836fc778b2e433e
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 4d48b14..aff11a8 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -6,9 +6,15 @@
 
 #include <boot_rom.h>
 #include <mem_io.h>
-#include <fs1_regs.h>
+#include <fs1_blocks_linux_driver.h>
 #include <fs1_reg_map.h>
+#include <arc_local_conf_regs.h>
+#include <arc_local_conf_masks.h>
+#include <psoc_boot_conf_regs.h>
+#include <psoc_security_regs.h>
+#include <global_conf_regs.h>
 #include <fs1_reg_mask.h>
+#include <fs1_fw_if.h>
 #include <status.h>
 
 uint8_t boot_rom_get_die_id(void)
@@ -33,23 +39,10 @@
 
 static void boot_rom_persist_data_non_zero_init(void)
 {
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].img_ptr_spi_offset =
-							HL_EROM_IMG_PTRS_BASE;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].dest_addr_dma =
-							HL_BOOT_PTR_EROM_DMA_DEST;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].dest_addr_arc =
-							HL_BOOT_PTR_EROM_DMA_DEST_ARC;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].boot_desc_flag = 0;
-
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].img_ptr_spi_offset =
-							HL_SPI_IMG_PTRS_BASE;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].dest_addr_dma =
-							HL_BOOT_PTR_SPI_DMA_DEST;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].dest_addr_arc =
-							HL_BOOT_PTR_SPI_DMA_DEST_ARC;
-	boot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].boot_desc_flag = 0;
+	/* TODO: init non zero data */
 }
 
+#define HL_ARC1_DCCM_PERSIST_SECTION_SIZE		(3 * 1024) /* 3KB */
 void boot_rom_persistent_dccm_init(uint8_t die_id)
 {
 	volatile uint32_t *dccm_addr = (uint32_t *) HL_DCCM_BASE;
